pkg/api/types.go

```diff
@@ -1803,6 +1811,15 @@ type SecretType string
 
 const (
 	SecretTypeOpaque SecretType = "Opaque" // Default; arbitrary user-defined data
+
+	// SecretTypeDockerConfig contains a dockercfg file that follows the same format rules as ~/.dockercfg
+	//
+	// Required fields:
+	// - Secret.Data["dockercfg"] - a serialized ~/.dockercfg file
```

Comment:
> it might clarify things if you put 'DockerConfigKey' instead of "dockercfg" as the key, since it will establish the linkage between the two constants.

I was trying to keep the name as close as possible to do the file that docker uses by default: `~/.dockercfg`.  If you're bundling a secret automatically via a helper command, stripping the leading '.' to make a dns1123 compatible name is natural.  That means that this secret could be made via: `bundle-secret ~/.dockercfg`.